include/deal.II/distributed/tria.h

```diff
@@ -1096,6 +1097,55 @@ namespace parallel
 #endif
 
 
+
+namespace internal
+{
+  namespace TriangulationImplementation
+  {
+    /**
+     * Communicate refinement flags on ghost cells from the owner of the cell.
+     *
+     * This is necessary to get consistent refinement, as mesh/ smoothing would
+     * undo some of the requested coarsening/refinement.
+     */
+    template <int dim, int spacedim>
+    void
+    exchange_refinement_flags(
+      dealii::parallel::distributed::Triangulation<dim, spacedim> &tria)
+    {
+      auto pack =
```

Comment:
This is an expensive function. It doesn't have to live in the `.h` file. Just forward declare it here and implement it in the .cc file.

Reply:
I moved it to the .cc file. Let me know if you want it placed somewhere else.